### PR TITLE
03 game author testing

### PR DESCRIPTION
**Pull request by gak92**
In this pull request:

- [X] Create test cases for the game class
- [X] Create test cases for the Author class

### CODE REVIEW

File: spec/author_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require 'date'
+require_relative 'spec_helper'
+
+describe Author do
```
Comment (Owusu-Desmond) [inline spec/author_spec.rb:4]: From the project requirement, we should test all implemented methods. Well done you have tested some of the `author` methods but not all, please kindly test the `add_author` and the `list_authors` methods in your `author_module.rb`

File: spec/author_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  describe '#items' do
+    it 'return the empty list of items' do
+      @author.items.should eql []
+    end
+  end
+
+  describe '#add_item' do
+    it 'return the lenth list of items' do
```
Comment (Owusu-Desmond) [inline spec/author_spec.rb]: - [ ] typo : `lenth` should be `length`

File: spec/game_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  describe '#multiplayer' do
+    it 'return the true for multiplayer' do
+      @game.multiplayer.should eql true
+    end
+  end
+
+  describe '#last_played_at' do
+    it 'return the true for last_played_at' do
+      @game.last_played_at.should eql Date.parse('2022-10-11')
+    end
```
Comment (Owusu-Desmond) [inline spec/game_spec.rb]: - [ ] Please kindly check the description of your `it` block, What you are expecting to be is not `true` but `date` so you can modify your description](https://github.com/Owusu-Desmond/catalog-of-my-things-app/pull/34#discussion_r993353078)

File: spec/game_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  describe '#publish_date' do
+    it 'return the true for publish_date' do
+      @game.publish_date.should eql '2019-11-11'
```
Comment (Owusu-Desmond) [inline spec/game_spec.rb]: - [ ] Please kindly check the description of your `it` block, What you are expecting to be `date` which is in a form of a string but your description is describing it to return a boolean which does not match. So you can modify your description

File: spec/game_spec.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+require 'date'
+require_relative 'spec_helper'
+
+describe Game do
+  before :each do
+    @game = Game.new(true, '2022-10-11', '2019-11-11')
+  end
```
Comment (Owusu-Desmond) [inline spec/game_spec.rb:8]: From the project requirement, we should test all implemented methods. Well done you have tested some of the `game` methods but not all, please kindly test the `add_game` and the `list_game` methods in your `game_module.rb`

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @gak92 👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Most associations working 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gak92** [comment]
@Owusu-Desmond,
Changes have been done and added the test cases for the remaining methods.

**Review by Owusu-Desmond — APPROVED** [review]
Hi @gak92 

Thank you for implementing all the changes required. Your task is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good design 💯 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Owusu-Desmond** [comment]
Hi @Mesi21, you need to add your review to this pull request

File: game/game_module.rb [hunk 6]
```diff
@@ -1,31 +1,21 @@
...
+    # save_game(new_game)
     @games.push({ 'multiplayer' => new_game.multiplayer, 'last_played_at' => new_game.last_played_at,
                   'publish_date' => new_game.publish_date })
+    'Game has been created successfully'
   end
 
+  # def save_game(new_game)
+  #   @games.push({ 'multiplayer' => new_game.multiplayer, 'last_played_at' => new_game.last_played_at,
+  #                 'publish_date' => new_game.publish_date })
+  # end
```
Comment (Owusu-Desmond) [inline game/game_module.rb:17]: Hi @gak92 

This is optional, you can remove this comment to make the code clean since it is no longer needed now

Comment (gak92) [reply game/game_module.rb:17]: @Owusu-Desmond,
Sure, I will do that.

**Review by Mesi21 — APPROVED** [review]
Hi @gak92 ,  

Thank you for implementing all the changes required. Your task is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good design 💯 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**